From c54393d14bc303d5b8cecda6971b312ffed36615 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Tue, 7 Apr 2015 13:12:55 -0700
Subject: [PATCH 101/176] PARQUET-234: Add ParquetInputSplit methods for
 compatibility.

Author: Ryan Blue <blue@apache.org>

Closes #159 from rdblue/PARQUET-234 and squashes the following commits:

b09d34d [Ryan Blue] PARQUET-234: Add ParquetInputSplit methods for compatibility.

Conflicts:
	parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java
Resolution:
    Replace fixes from c89be03 with upstream fixes.
---
 .../java/parquet/hadoop/ParquetInputSplit.java     |   28 +++++++++++---------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java
index c882d8d..4635098 100644
--- a/parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java
+++ b/parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java
@@ -121,7 +121,8 @@ public class ParquetInputSplit extends FileSplit implements Writable {
    */
   @Deprecated
   public List<BlockMetaData> getBlocks() {
-    return Arrays.asList();
+    throw new UnsupportedOperationException(
+        "Splits no longer have row group metadata, see PARQUET-234");
   }
 
   /**
@@ -186,21 +187,23 @@ public class ParquetInputSplit extends FileSplit implements Writable {
   }
 
   /**
-   * @return the file schema
+   * @return the requested schema
    * @deprecated the file footer is no longer read before creating input splits
    */
   @Deprecated
-  public String getFileSchema() {
-    return null;
+  String getRequestedSchema() {
+    throw new UnsupportedOperationException(
+        "Splits no longer have the requested schema, see PARQUET-234");
   }
 
   /**
-   * @return the requested schema
-   * @deprecated read schema is no longer passed via an input split
+   * @return the file schema
+   * @deprecated the file footer is no longer read before creating input splits
    */
   @Deprecated
-  public String getRequestedSchema() {
-    return null;
+  public String getFileSchema() {
+    throw new UnsupportedOperationException(
+        "Splits no longer have the file schema, see PARQUET-234");
   }
 
   /**
@@ -216,16 +219,17 @@ public class ParquetInputSplit extends FileSplit implements Writable {
    */
   @Deprecated
   public Map<String, String> getExtraMetadata() {
-    return new HashMap<String, String>();
+    throw new UnsupportedOperationException(
+        "Splits no longer have file metadata, see PARQUET-234");
   }
 
   /**
    * @return app specific metadata provided by the read support in the init phase
-   * @deprecated read metadata is no longer determined on the client side
    */
   @Deprecated
-  public Map<String, String> getReadSupportMetadata() {
-    return new HashMap<String, String>();
+  Map<String, String> getReadSupportMetadata() {
+    throw new UnsupportedOperationException(
+        "Splits no longer have read-support metadata, see PARQUET-234");
   }
 
   /**
-- 
1.7.9.5

